src/Microsoft.AspNetCore.Mvc.Core/Internal/AcceptHeaderParser.cs

```diff
@@ -18,6 +18,19 @@ public static IList<MediaTypeSegmentWithQuality> ParseAcceptHeader(IList<string>
             return parsedValues;
         }
 
+        public static bool TryParseAcceptHeader(IList<string> acceptHeaders, IList<MediaTypeSegmentWithQuality> values)
+        {
+            try
+            {
+                ParseAcceptHeader(acceptHeaders, values);
```

Comment:
Suggest creating a `TryParseAcceptHeader()` method that takes an `int out invalidCharIndex` parameter and looks a lot like existing `ParseAcceptHeader()`. Then use that here (ignoring the `out` parameter) and in a short wrapper for the existing `ParseAcceptHeader()`. Would avoid a "try" method that doesn't try at all -- just fails and recovers.